pkg: gecko_sdk: can't build with LLVM

#### Description
When testing for #9398 locally I wasn't able to build any board that uses `gecko_sdk` with LLVM/clang, but this should be possible, because AFAIK they are Cortex-M based.

#### Steps to reproduce the issue
Try building any application for a board using `gecko_sdk` using LLVM/clang, it will fail on an "unknown warning option":

```
$ TOOLCHAIN=llvm BOARD=ikea-tradfri make -C examples/hello-world/
make: Entering directory '/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world'
make[1]: Nothing to be done for 'Makefile.include'.
Building application "hello-world" for "ikea-tradfri" with MCU "efm32".

rm -Rf /home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/pkg/ikea-tradfri/gecko_sdk
mkdir -p /home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/pkg/ikea-tradfri/gecko_sdk
/home/mlenders/Repositories/RIOT-OS/RIOT/dist/tools/git/git-cache clone "https://github.com/basilfx/RIOT-gecko-sdk" "a5e7be18ef3d7947ab6f04f9cd74f8808da7aece" "/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/pkg/ikea-tradfri/gecko_sdk"
git-cache: cloning from cache. tag=commita5e7be18ef3d7947ab6f04f9cd74f8808da7aece
touch /home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/pkg/ikea-tradfri/gecko_sdk/.git-downloaded
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/pkg/gecko_sdk
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/pkg/ikea-tradfri/gecko_sdk/dist
"make" -C /home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/pkg/ikea-tradfri/gecko_sdk/dist/emlib-extra/src
error: unknown warning option '-Wno-int-in-bool-context'; did you mean '-Wno-gnu-include-next'? [-Werror,-Wunknown-warning-option]
/home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base:83: recipe for target '/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/ikea-tradfri/gecko_sdk_emlib_extra/em_usart_utils.o' failed
make[3]: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/ikea-tradfri/gecko_sdk_emlib_extra/em_usart_utils.o] Error 1
/home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base:20: recipe for target 'ALL--/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/pkg/ikea-tradfri/gecko_sdk/dist/emlib-extra/src' failed
make[2]: *** [ALL--/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/pkg/ikea-tradfri/gecko_sdk/dist/emlib-extra/src] Error 2
Makefile:15: recipe for target 'all' failed
make[1]: *** [all] Error 2
/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/../../Makefile.include:459: recipe for target '/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/ikea-tradfri/gecko_sdk.a' failed
make: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world/bin/ikea-tradfri/gecko_sdk.a] Error 2
make: Leaving directory '/home/mlenders/Repositories/RIOT-OS/RIOT/examples/hello-world'
```

#### Expected results
Should build with LLVM.

#### Actual results
Build fails due to an unknown warning option

#### Versions
```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial
$ ./dist/tools/ci/print_toolchain_versions.sh 
Installed compiler toolchains 
-----------------------------
             native gcc: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
      arm-none-eabi-gcc: arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 7-2018-q3-update) 7.3.1 20180622 (release) [ARM/embedded-7-branch revision 261907]
                avr-gcc: avr-gcc (GCC) 4.9.2
       mips-mti-elf-gcc: mips-mti-elf-gcc (Codescape GNU Tools 2016.05-03 for MIPS MTI Bare Metal) 4.9.2
             msp430-gcc: msp430-gcc (GCC) 4.6.3 20120301 (mspgcc LTS 20120406 unpatched)
   riscv-none-embed-gcc: riscv-none-embed-gcc (GNU MCU Eclipse RISC-V Embedded GCC, 64-bits) 7.2.0
                  clang: clang version 3.8.0-2ubuntu4 (tags/RELEASE_380/final)

Installed compiler libs
-----------------------
   arm-none-eabi-newlib: "3.0.0"
    mips-mti-elf-newlib: "2.1.0"
riscv-none-embed-newlib: "2.5.0"
               avr-libc: "1.8.0svn" ("20111229")

Installed development tools
---------------------------
                  cmake: cmake version 3.5.1
               cppcheck: Cppcheck 1.72
                doxygen: 1.8.11
                 flake8: 2.5.4 (pep8: 1.7.0, pyflakes: 1.1.0, mccabe: 0.2.1) CPython 3.5.2 on Linux
                    git: git version 2.7.4
             coccinelle: spatch version 1.0.4 with Python support and with PCRE support
```

--- a/pkg/gecko_sdk/Makefile
+++ b/pkg/gecko_sdk/Makefile
@@ -1,6 +1,6 @@
 PKG_NAME=gecko_sdk
 PKG_URL=https://github.com/basilfx/RIOT-gecko-sdk
-PKG_VERSION=a5e7be18ef3d7947ab6f04f9cd74f8808da7aece
+PKG_VERSION=6f076bc0557e6f9a44a9d08e0a734f0d24d7c97c
 PKG_LICENSE=Zlib
 
 ifneq ($(CPU),efm32)

--- a/pkg/gecko_sdk/Makefile
+++ b/pkg/gecko_sdk/Makefile
@@ -7,7 +7,9 @@ ifneq ($(CPU),efm32)
   $(error This package can only be used with EFM32 CPUs)
 endif
 
-CFLAGS += -Wno-int-in-bool-context
+ifneq (llvm,$(TOOLCHAIN))
+  CFLAGS += -Wno-int-in-bool-context
+endif
 
 .PHONY: all
 

